### PR TITLE
Get rid of shareItem

### PR DESCRIPTION
**Pull request by MorrisJobke**
@rullzer Mind to look into how to do the stuff with the new sharing code, because as of now it's only used by tests ;)

cc @nextcloud/sharing

### CODE REVIEW

**Comment by rullzer** [comment]
The tagmanager stuff is little bit trickier to rip out. This is because 'in theory' this was designed to be able to tag everything. And of course sharing of tags was done via the share table :S

However, looking at the code it seems that we only use it for files... but still. some proper cleanup here is needed I think.

**Comment by MorrisJobke** [comment]
Moved most of this to #9564

**Comment by MorrisJobke** [comment]
:+1: from my side for the code by @rullzer

File: lib/private/Share/Share.php [hunk 1]
```diff
@@ -327,6 +327,7 @@ public static function getItemShared($itemType, $itemSource, $format = self::FOR
 	 * @throws \OC\HintException when the share type is remote and the shareWith is invalid
 	 * @throws \Exception
 	 * @since 5.0.0 - parameter $itemSourceName was added in 6.0.0, parameter $expirationDate was added in 7.0.0, parameter $passwordChanged added in 9.0.0
+	 * TESTS ONLY
```
Comment (blizzz) [inline lib/private/Share/Share.php]: ?

Comment (blizzz) [reply lib/private/Share/Share.php]: perhaps a deprecated tag to make it more clear. nitpicking.

**Review by blizzz — APPROVED** [review]

File: lib/private/Share/Share.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/private/Share/Share.php]: Just a note that this method is only used by tests. Should I rephrase it a bit?

Comment (blizzz) [reply lib/private/Share/Share.php]: ok for a private class, don't bother

**Review by rullzer — APPROVED** [review]
:elephant:

**Comment by MorrisJobke** [comment]
Only PHP doc changed -> merge it

**Pull request merged**